### PR TITLE
Story/project scorecard discrepancies 177195362

### PR DESCRIPTION
**Pull request by awisema**

### CODE REVIEW

File: drivers/hud_apr/app/models/hud_apr/generators/shared/fy2020/question_six.rb [hunk 1]
```diff
@@ -393,13 +393,13 @@ def self.question_number
...
-        a_t[:income_date_at_start].eq(nil).
-          or(a_t[:income_date_at_start].not_eq(a_t[:first_date_in_program])).
-          or(a_t[:income_from_any_source_at_start].in([nil, 8, 9])).
-          or(a_t[:income_from_any_source_at_start].eq(0).
-            and(income_jsonb_clause(1, a_t[:income_sources_at_start].to_sql))).
-          or(a_t[:income_from_any_source_at_start].eq(1).
-            and(income_jsonb_clause(1, a_t[:income_sources_at_start].to_sql, negation: true))),
+        a_t[:income_date_at_annual_assessment].eq(nil).
+          or(a_t[:annual_assessment_in_window].eq(false)).
+          or(a_t[:income_from_any_source_at_annual_assessment].in([nil, 8, 9])).
```
Comment (eanders) [inline drivers/hud_apr/app/models/hud_apr/generators/shared/fy2020/question_six.rb]: Does this correctly turn into `income_from_any_source_at_annual_assessment in (8, 9) or income_from_any_source_at_annual_assessment is null`?

Comment (awisema) [reply drivers/hud_apr/app/models/hud_apr/generators/shared/fy2020/question_six.rb]: Good catch! `leavers.where(a_t[:income_from_any_source_at_exit].eq(nil)).count != leavers.where(a_t[:income_from_any_source_at_exit].in([nil])).count)`

File: drivers/hud_apr/app/models/hud_apr/generators/shared/fy2020/question_six.rb [hunk 2]
```diff
@@ -258,7 +258,8 @@ def self.question_number
       # veteran status
       answer = @report.answer(question: table_name, cell: 'B2')
       members = universe.members.where(
-        a_t[:veteran_status].in([nil, 8, 9]).
+        a_t[:veteran_status].in([8, 9]).
+          or(a_t[:veteran_status].eq(nil)).
           or(a_t[:veteran_status].eq(1).
```
Comment (eanders) [inline drivers/hud_apr/app/models/hud_apr/generators/shared/fy2020/question_six.rb]: Why is this looking at 8, 9, 1 or nil?, isn't that the same as any veteran status?

File: drivers/hud_apr/app/models/hud_apr/generators/shared/fy2020/question_six.rb [hunk 3]
```diff
@@ -393,13 +399,14 @@ def self.question_number
...
-          or(a_t[:income_from_any_source_at_start].in([nil, 8, 9])).
-          or(a_t[:income_from_any_source_at_start].eq(0).
-            and(income_jsonb_clause(1, a_t[:income_sources_at_start].to_sql))).
-          or(a_t[:income_from_any_source_at_start].eq(1).
-            and(income_jsonb_clause(1, a_t[:income_sources_at_start].to_sql, negation: true))),
+        a_t[:income_date_at_annual_assessment].eq(nil).
+          or(a_t[:annual_assessment_in_window].eq(false)).
+          or(a_t[:income_from_any_source_at_annual_assessment].in([8, 9])).
+          or(a_t[:income_from_any_source_at_annual_assessment].eq(nil)).
+          or(a_t[:income_from_any_source_at_annual_assessment].eq(0).
```
Comment (eanders) [inline drivers/hud_apr/app/models/hud_apr/generators/shared/fy2020/question_six.rb:404]: Why not include 0 with 8, 9?

File: drivers/hud_apr/app/models/hud_apr/generators/shared/fy2020/question_six.rb (hunk 2 above)
Comment (awisema) [reply drivers/hud_apr/app/models/hud_apr/generators/shared/fy2020/question_six.rb]: I'll adjust the formatting to make it clearer -- but this is either [nil, 8, 9], or 1 and age < 18

Comment (eanders) [reply drivers/hud_apr/app/models/hud_apr/generators/shared/fy2020/question_six.rb]: Got it!

File: drivers/hud_apr/app/models/hud_apr/generators/shared/fy2020/question_six.rb (hunk 3 above)
Comment (awisema) [reply drivers/hud_apr/app/models/hud_apr/generators/shared/fy2020/question_six.rb:404]: Same deal -- this is 'any source' says no, but then there is a source. Will add comments

Comment (eanders) [reply drivers/hud_apr/app/models/hud_apr/generators/shared/fy2020/question_six.rb:404]: Ah, the trailing and.

**Pull request merged**